src/triple.rs

```diff
@@ -17,41 +17,40 @@ pub enum Endianness {
     Big,
 }
 
-/// The width of a pointer (in the default address space).
+/// The size in bits of a type.
 #[derive(Copy, Clone, Debug, PartialEq, Eq, Hash)]
 #[allow(missing_docs)]
-pub enum PointerWidth {
+pub enum Size {
     U16,
     U32,
     U64,
 }
```

Comment:
I'd like to propose we introduce a new type rather than repurposing `PointerWidth`. `PointerWidth` is one of the few non-`non_exhaustive` enums in target-lexicon, and it's convenient for the very common case of just asking the library "is this platform 32-bit or 64-bit (or 16-bit)?" since that bit of information is needed in a lot of places.

But for C type layout, as we look forward to adding support for struct layout, but also SIMD types, minifloats, `int128`, _Complex, long double, _Complex long double, and oh, the places we'll go ;-), we'll want something to extend to new uses. So I suggest a new `Size` struct with a private integer field here, which can hold any size.

Reply:
I'm happy to make a new enum, I felt a little uncomfortable removing `non_exhaustive` anyway.

> I suggest a new Size struct with a private integer field here, which can hold any size.

Ok, if we made it a struct I could also add `char_bit` which would make `bytes()` more useful. I think the `bytes()` and `bits()` API is still useful, I'd like to keep it.